crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -0,0 +1,1006 @@
+use super::{block, flatten, integer, node, optim, ssa_form};
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::io;
+
+use super::super::environment::Environment;
+use super::super::errors::{RuntimeError, RuntimeErrorKind};
+use crate::object::Object;
+use crate::ssa::acir_gen::Acir;
+use crate::ssa::node::Node;
+use crate::Evaluator;
+use acvm::FieldElement;
+use arena;
+use noirc_frontend::hir::Context;
+use noirc_frontend::hir_def::function::HirFunction;
+use noirc_frontend::hir_def::{
+    expr::{
+        HirBinaryOp, HirBinaryOpKind, HirBlockExpression, HirCallExpression, HirExpression,
+        HirForExpression, HirLiteral,
+    },
+    stmt::{HirConstrainStatement, HirLetStatement, HirPrivateStatement, HirStatement},
+};
+use noirc_frontend::node_interner::{ExprId, FuncId, IdentId, StmtId};
+//use noirc_frontend::{FunctionKind, Type};
+use num_bigint::BigUint;
+
+// This is a 'master' class for generating the SSA IR from the AST
+// It contains all the data; the node objects representing the source code in the nodes arena
+// and The CFG in the blocks arena
+// everything else just reference objects from these two arena using their index.
+pub struct IRGenerator<'a> {
+    pub context: Option<&'a Context>,
+
+    pub first_block: arena::Index,
+    pub current_block: arena::Index,
+    pub blocks: arena::Arena<block::BasicBlock>,
+    pub nodes: arena::Arena<node::NodeObj>,
+    pub id0: arena::Index, //dummy index.. should we put a dummy object somewhere?
+    pub value_name: HashMap<arena::Index, u32>,
+    pub sealed_blocks: HashSet<arena::Index>,
+}
+
+impl<'a> IRGenerator<'a> {
+    pub fn new(context: &Context) -> IRGenerator {
+        let mut pc = IRGenerator {
+            context: Some(context),
+            id0: IRGenerator::dummy_id(),
+            first_block: IRGenerator::dummy_id(),
+            current_block: IRGenerator::dummy_id(),
+            blocks: arena::Arena::new(),
+            nodes: arena::Arena::new(),
+            // dummy_instruction: ParsingContext::dummy_id(),
+            value_name: HashMap::new(),
+            sealed_blocks: HashSet::new(),
+        }; //, objects: arena::Arena::new()
+        block::create_first_block(&mut pc);
+        pc
+    }
+
+    fn print_operand(&self, idx: arena::Index) -> String {
```

Comment:
`operand_to_string` may be better here since the operand isn't being printed

Reply:
I renamed into 'to_string', because operand is relevant for the caller, not for the function itself.